concepts/microservices/1-introduction.md

```diff
@@ -0,0 +1,92 @@
+# Introduction to Microservices
+
+## Outline
+
+- [What Are Microservices](#what-are-microservices)
+- [Why Microservices](#why-microservices)
+- [Benefits of Microservices](#benefits-of-microservices)
+  - [Heterogeneity](#heterogeneity)
+  - [Resilience](#resilience)
+  - [Scaling](#scaling)
+  - [Organizational Alignment](#organizational-alignment)
+  - [Composability](#composability)
+
+## What are Microservices
+
+Microservices are small, autonomous services that work together in tandem.
+There are several principles that a microservice adheres to:
+
+- Single Responsibility
+- Autonomous
+- Maintainable
+
+A microservice embodies the Single Responsibility Principle (SRP) - they are focused on
+only one thing, and doing it well. They are also autonomous and maintainable.
+
+An important question to answer is _how small_ should microservices be? There's
+no set standard for this question, but a good rule of thumb for size is that
+it could be refactored in 2 weeks. The smaller the service, the more you're
+able to maximize the benefits of microservices. The trade-off with smaller
+services is that you have to then manage the interdependence of small
+services with others, which leads to more moving parts in your system.
+
+In short, microservices should aim to be autonomous and maintainable.
+Communication with other services should happen via network calls in order
+to better enforce separation and prevent tight coupling.
+
+## Why Microservices
+
+Where did microservices come from? In short, they came about as a result of
+rapid iteration and advancement in how engineers manage their software.
+Microservices have emerged With the rise of domain-driven design, continuous
+delivery, virtualization, and automation tooling.
+
+By embracing the principles around microservices, engineers can deliver
+software much more quickly and reliably. This gives engineers more freedom
+to react and make decisions independently of other groups / parties / code
+boundaries.
+
+## Benefits of Microservices
```

Comment:
Are there any cons, or when do we better not to choose microservices architecture?

Reply:
This will come in a separate section. This is more an overview of what to consider for microservices.